src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.VariableIdentifier.cs

```diff
@@ -26,26 +27,60 @@ public bool Exists
             {
                 get { return (object)Symbol != null; }
             }
+
             public override int GetHashCode()
             {
-                return Roslyn.Utilities.Hash.Combine(Symbol?.OriginalDefinition, ContainingSlot);
+                Debug.Assert(Exists);
+
+                int currentKey = ContainingSlot;
+                // MemberIndexOpt, if available, is a cheap approach to comparing relative members.
+                int? thisIndex = Symbol.MemberIndexOpt;
+                return thisIndex.HasValue ?
```

Comment:
>thisIndex.HasValue [](start = 23, length = 18)

is the conditional inverted?

Reply:
Yes, thanks.

---
In reply to: [259136898](https://github.com/dotnet/roslyn/pull/33562#discussion_r259136898) [](ancestors = 259136898)